kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/BatchesResource.scala

```diff
@@ -332,6 +339,12 @@ private[v1] class BatchesResource extends ApiRequestContext with Logging {
                 Some(userName),
                 // prevent that the batch be marked as terminated if application state is NOT_FOUND
                 Some(metadata.engineOpenTime).filter(_ > 0).orElse(Some(System.currentTimeMillis)))
+              // if the batch app is terminated, update the metadata in db.
+              if (BatchJobSubmission.applicationTerminated(batchAppStatus)) {
+                sessionManager.updateMetadata(Metadata(
+                  identifier = batchId,
+                  engineState = batchAppStatus.get.state.toString))
```

Comment:
should we update all the batchAppStatus into metadata?

Including appId, name, state, error.

Reply:
+1